src/ai/backend/manager/repositories/scaling_group/creators.py

```diff
@@ -0,0 +1,41 @@
+from __future__ import annotations
+
+from collections.abc import Mapping
+from dataclasses import dataclass, field
+from typing import Any, Optional, override
+
+from ai.backend.manager.models.scaling_group import ScalingGroupOpts, ScalingGroupRow
+from ai.backend.manager.repositories.base.creator import CreatorSpec
+
+
+@dataclass
+class ScalingGroupCreatorSpec(CreatorSpec[ScalingGroupRow]):
+    """CreatorSpec for scaling group."""
+
+    name: str
+    driver: str
+    scheduler: str
+    description: Optional[str] = None
+    is_active: bool = True
+    is_public: bool = True
+    wsproxy_addr: Optional[str] = None
+    wsproxy_api_token: Optional[str] = None
+    driver_opts: Mapping[str, Any] = field(default_factory=dict)
+    scheduler_opts: Optional[ScalingGroupOpts] = None
+    use_host_network: bool = False
+
+    @override
+    def build_row(self) -> ScalingGroupRow:
+        row = ScalingGroupRow()
+        row.name = self.name
+        row.description = self.description
+        row.is_active = self.is_active
+        row.is_public = self.is_public
+        row.wsproxy_addr = self.wsproxy_addr
+        row.wsproxy_api_token = self.wsproxy_api_token
+        row.driver = self.driver
+        row.driver_opts = dict(self.driver_opts)
+        row.scheduler = self.scheduler
+        row.scheduler_opts = self.scheduler_opts if self.scheduler_opts else ScalingGroupOpts()
+        row.use_host_network = self.use_host_network
```

Comment:
Please create it with a constructor. It is better to pass the constructor after all values are prepared, and I don't think it's good to create the constructor with empty values and then add arguments.